flink-runtime/src/main/java/org/apache/flink/runtime/metrics/TimerGauge.java

```diff
@@ -47,33 +47,36 @@ public TimerGauge(Clock clock) {
     }
 
     public synchronized void markStart() {
-        if (currentMeasurementStart == 0) {
-            currentMeasurementStart = clock.absoluteTimeMillis();
-        }
+        markStartUnsafe(clock.absoluteTimeMillis());
     }
 
     public synchronized void markEnd() {
-        if (currentMeasurementStart != 0) {
-            currentCount += clock.absoluteTimeMillis() - currentMeasurementStart;
-            currentMeasurementStart = 0;
-        }
+        markEndUnsafe(clock.absoluteTimeMillis());
```

Comment:
Maybe revert those changes after all for the same reason as:
https://github.com/apache/flink/pull/16772#discussion_r688401559
?